master/internal/experiment/bulk_action.go

```diff
@@ -734,3 +735,104 @@ func MoveExperiments(ctx context.Context,
 	}
 	return results, nil
 }
+
+func changeExperimentConfigLogRetention(ctx context.Context, database db.DB,
+	expID int, numDays int16,
+) error {
+	exp, err := db.ExperimentByID(ctx, expID)
+	if err != nil {
+		return errors.Wrap(err, "fetching experiment from database")
+	}
+
+	// In long experiments like asha, if the user called this api right after creating the experiment,
+	// only the trials that were created at that time have the log retention days changed, any trials
+	// created after a while will have the same log retention days as in the task spec. After some
+	// discussion, we came to the conclusion that it would be simplest to not allow the users to run
+	// this command before an experiment is completed.
+	if !model.TerminalStates[exp.State] {
+		return status.Error(codes.FailedPrecondition, fmt.Sprintf(
+			"experiment in non terminal state '%s', try again later", exp.State))
+	}
+
+	activeConfig, err := database.ActiveExperimentConfig(exp.ID)
+	if err != nil {
+		return errors.Wrapf(
+			err, "unable to load config for experiment %v", exp.ID,
+		)
+	}
+	activeConfig.SetLogRetentionDays(&numDays)
+
+	if err := database.SaveExperimentConfig(exp.ID, activeConfig); err != nil {
+		return errors.Wrapf(err, "patching experiment %d", exp.ID)
+	}
+
+	return nil
+}
+
+// RetainLogs retains logs for the given list of experiments.
+func RetainLogs(ctx context.Context, database db.DB,
```

Comment:
nit: would it make sense to rename this something like `BulkUpdateLogRentention`?

Reply:
done!